errbot/err.py

```diff
@@ -154,29 +156,39 @@ def main():
-    parser.add_argument('-l', '--list', action='store_true', help='Lists all the backends found.')
-
-    from .version import VERSION
-    parser.add_argument('-v', '--version', action='version', version='Err version {}'.format(VERSION))
-
-    backend_group = parser.add_mutually_exclusive_group()
-    backend_group.add_argument('-X', '--xmpp', action='store_true', help='XMPP backend [DEFAULT]')
-    backend_group.add_argument('-H', '--hipchat', action='store_true', help='Hipchat backend')
-    backend_group.add_argument('-I', '--irc', action='store_true', help='IRC backend')
-    backend_group.add_argument('-S', '--slack', action='store_true', help='Slack backend')
-    backend_group.add_argument('-T', '--text', action='store_true', help='locale text debug backend')
-    backend_group.add_argument('-G', '--graphic', action='store_true', help='local graphical debug mode backend')
-    backend_group.add_argument('-N', '--null', action='store_true', help='no backend')
+
+    mode_selection = parser.add_mutually_exclusive_group()
+    mode_selection.add_argument('-v', '--version', action='version', version='Errbot version {}'.format(VERSION))
+    mode_selection.add_argument('-r', '--restore', nargs='?', default=None, const='default',
+                                help='restore a bot from backup.py (default: backup.py from the bot data directory)')
+    mode_selection.add_argument('-l', '--list', action='store_true', help='list all available backends')
+    mode_selection.add_argument('--new-plugin', nargs='?', default=None, const='current_dir',
+                                help='create a new plugin in the specified directory')
+    mode_selection.add_argument('-T', '--text', dest="backend", action='store_const', const="Text",
+                                help='force local text backend')
+    mode_selection.add_argument('-G', '--graphic', dest="backend", action='store_const', const="Graphic",
+                                help='force local graphical backend')
 
     if not ON_WINDOWS:
-        option_group = parser.add_argument_group('arguments to run it as a Daemon')
+        option_group = parser.add_argument_group('optional daemonization arguments')
         option_group.add_argument('-d', '--daemon', action='store_true', help='Detach the process from the console')
         option_group.add_argument('-p', '--pidfile', default=None,
                                   help='Specify the pid file for the daemon (default: current bot data directory)')
 
     args = vars(parser.parse_args())  # create a dictionary of args
+
+    # This must come BEFORE the config is loaded below, to avoid printing
+    # logs as a side effect of config loading.
+    if args['new_plugin']:
+        directory = os.getcwd() if args['new_plugin'] == "current_dir" else args['new_plugin']
+        logging.getLogger('').setLevel(level=1000)  # Avoid logging stuff
```

Comment:
Maybe something like that ?
logging.removehandler(handler) for handler in logging.handlers